### DISCUSSION TITLE
addControl in a defined DIV

### DISCUSSION THREAD

**Discussion by sylvain258** [maplibre/maplibre-gl-js#2805] [Q&A]
Hello,

I found few layer trees on github for MapBox/MapLibre. I have a problem about positions of panels. Effect, addControl asks a parameter 'position' defined with 4 values ('top-left', 'top-right', 'bottom-left', 'bottom-right'). 

In my mapping app, I have a sidebar with tabs. 

I want put this addControl in a div inside a tab. 

I would like to know if it's possible.

Thanks

Sylvain 

Example Mapbox GL JS Layer-Tree from Gartrell-Group Github. 
![Capture d’écran du 2023-07-05 22-03-53](https://github.com/maplibre/maplibre-gl-js/assets/33748168/0f1576f0-1a6a-4e24-a8ce-4720020f0971)

**Comment by HarelM** [comment]
I don't think it is.
But you can inherit from `IControl` and use the map API basically.
You can also probably look at the relevant control code and hack it to be elesewhere or simply get the gist of it and write it in your own component.
